build-support/bin/shellcheck.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import shutil
+import subprocess
+from glob import glob
+
+from common import die, green
+
+
+def main() -> None:
+  ensure_shellcheck_installed()
+  run_shellcheck()
+
+
+def ensure_shellcheck_installed() -> None:
+  if shutil.which("shellcheck") is None:
+    die("`shellcheck` not installed! You may download this through your operating system's "
+        "package manager, such as brew, apt, or yum. See "
+        "https://github.com/koalaman/shellcheck#installing.")
+
+
+def run_shellcheck() -> None:
+  targets = glob("./build-support/bin/**/*.sh", recursive=True) + [
```

Comment:
Could we do something like `rg -l "#!/bin/bash" build-support` to get an automatic list of these?

Reply:
I think this is a great suggestion so looked into it and run into a couple issues:
1) Not every environment has ripgrep installed, including CI.
    * Using something portable like `find` + `awk` is much harder to read.
    * `find` also doesn't skip our `.gitignore` by default, which is not good.
    * We can certainly require they install it—that's fine by me. But it would mean to run `githooks/pre-commit` people now need both `ripgrep` and `shellcheck` installed.
2) Ripgrep doesn't have a way to search only the first line, at least that I know of. 
     * There was a Python file with this shebang in it that was a false positive.
3) Not everything uses the correct shebang `#!/usr/bin/env bash`, some use `#!/bin/bash`.
    * I'll actually fix this in a dedicated PR!

I think all of these are solvable. Are we okay with #1 to require ripgrep to be installed? I'd love if we could make that assumption, then we could start using ripgrep in other scripts!

Reply:
Actually #1 is the one I'm least comfortable with, unless we somehow get a ripgrep binary at the bootstrap step (same way we fetch [the rust toolchain](https://github.com/pantsbuild/pants/blob/5eed2e7be5aa4672485f06e043bcec24bb9c668d/build-support/bin/native/bootstrap_rust.sh#L29), although it's definitely not my area of expertise). Although at this point this definitely becomes a separate PR, along with #2 and maybe #3 if noone has gotten around to it yet.

Reply:
So fine with hardcoded paths by now.